sycl/test-e2e/NonUniformGroups/ballot_group_algorithms.cpp

```diff
@@ -79,6 +79,10 @@ int main() {
           bool AnyResult = sycl::any_of_group(BallotGroup, (LID == 0));
           AnyAcc[WI] = (AnyResult == true);
 
+          // Note that this barrier is required for the test to pass for
+          // cuda 12.4 even for sm_60 and later devices. This appears to be a
+          // cuda ptxas bug.
+          sycl::group_barrier(BallotGroup);
```

Comment:
Could we maybe mask it with `#ifdef __NVPTX__` so we don't inadvertently ignore problems on other targets?

Reply:
I also wonder if it should be the fix in the implementation and not in the test.

Reply:
Yeah makes sense. Ta

Reply:
> I also wonder if it should be the fix in the implementation and not in the test.

I can create a ticket to confirm that it is a ptx bug and report it to nvidia. They also just released cuda 12.5, so it is possible it is working there. Prior to cuda 12.4 this test passed.